aws/resource_aws_guardduty_publishing_destination_test.go

```diff
@@ -0,0 +1,273 @@
+	resource.AddTestSweepers("aws_guardduty_publishing_destination", &resource.Sweeper{
+		Name: "aws_guardduty_publishing_destination",
+		F:    testSweepGuarddutyPublishingDestinations,
+	})
+}
+
+func testSweepGuarddutyPublishingDestinations(region string) error {
+	client, err := sharedClientForRegion(region)
+
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+
+	conn := client.(*AWSClient).guarddutyconn
+	var sweeperErrs *multierror.Error
+
+	detect_input := &guardduty.ListDetectorsInput{}
+
+	err = conn.ListDetectorsPages(detect_input, func(page *guardduty.ListDetectorsOutput, lastPage bool) bool {
+		for _, detectorID := range page.DetectorIds {
+			list_input := &guardduty.ListPublishingDestinationsInput{
+				DetectorId: detectorID,
+			}
+
+			err = conn.ListPublishingDestinationsPages(list_input, func(page *guardduty.ListPublishingDestinationsOutput, lastPage bool) bool {
+				for _, destination_element := range page.Destinations {
+					input := &guardduty.DeletePublishingDestinationInput{
+						DestinationId: destination_element.DestinationId,
+						DetectorId:    detectorID,
+					}
+
+					log.Printf("[INFO] Deleting GuardDuty Publish Destination: %s", *destination_element.DestinationId)
+					_, err := conn.DeletePublishingDestination(input)
+
+					if err != nil {
+						sweeperErr := fmt.Errorf("error deleting GuardDuty Pusblish Destination (%s): %w", *destination_element.DestinationId, err)
+						log.Printf("[ERROR] %s", sweeperErr)
+						sweeperErrs = multierror.Append(sweeperErrs, sweeperErr)
+					}
+				}
+				return !lastPage
+			})
+		}
+		return !lastPage
+	})
+
+	if err != nil {
+		sweeperErr := fmt.Errorf("Error receiving Guardduty detectors for publish sweep : %w", err)
+		log.Printf("[ERROR] %s", sweeperErr)
+		sweeperErrs = multierror.Append(sweeperErrs, sweeperErr)
+	}
+
+	if testSweepSkipSweepError(err) {
+		log.Printf("[WARN] Skipping GuardDuty Publish Destination sweep for %s: %s", region, err)
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error retrieving GuardDuty Publish Destinations: %s", err)
+	}
+
+	return sweeperErrs.ErrorOrNil()
+}
+
+func TestAccAwsGuardDutyPublishingDestination_basic(t *testing.T) {
+	resourceName := "aws_guardduty_publishing_destination.test"
+	bucketName := fmt.Sprintf("tf-test-%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsGuardDutyPublishingDestinationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsGuardDutyPublishingDestinationConfig_basic(bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsGuardDutyPublishingDestinationExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "detector_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "destination_arn"),
+					resource.TestCheckResourceAttrSet(resourceName, "kms_key_arn"),
```

Comment:
can we use `TestCheckResourceAttrPair` here instead?

Reply:
Refactored